virtool/samples/data.py

```diff
@@ -187,6 +191,64 @@ async def get(self, sample_id: str) -> Sample:
 
         return Sample(**document)
 
+    async def finalize(
+        self,
+        sample_id: str,
+        quality: Dict[str, Any],
+        _run_in_thread: callable,
+        data_path: Path,
+    ) -> Dict[str, Any]:
+        """
+        Finalize a sample document by setting a ``quality`` field
+        and ``ready`` to ``True``
+
+        :param sample_id: the id of the sample
+        :param quality: a dict contains quality data
+        :param _run_in_thread: the application thread running function
+        :param data_path: the application data path settings
```

Comment:
There may be a `self._config` attribute  which is the preferred way to access configuration.